src/tests/ftest/control/dmg_telemetry_io_latency.py

```diff
@@ -339,6 +354,81 @@ def test_io_latency_telmetry_metrics(self):
+        """Verify IO dtx metrics after running IOR.
+
+        Args:
+            metrics_data (dict): dictionary of io dtx "committed" metrics
+            test_metrics (dict): io dtx "committed" telemetry metrics
+            transfer_size(str): transfer_size used with ior
+
+        Returns:
+            status: (bool) True if metrics are verified for transfer size
+
+        """
+        block_size = self.params.get("block_size", "/run/*")
+        metrics = {}
+        for test_metric in test_metrics:
+            metrics[test_metric] = 0
+            for host in self.hostlist_servers:
+                # test assumes one engine per host
+                for rank in self.server_managers[-1].get_host_ranks([host]):
+                    for target in range(self.server_managers[-1].get_config_value("targets")):
+                        value = metrics_data[test_metric][host][str(rank)][str(target)][size]
+                        metrics[test_metric] = metrics[test_metric] + value
+        value = metrics["engine_io_dtx_committed"]
+        min_value = metrics["engine_io_dtx_committed_min"]
+        max_value = metrics["engine_io_dtx_committed_max"]
+        mean_value = metrics["engine_io_dtx_committed_mean"]
+        stddev_value = metrics["engine_io_dtx_committed_stddev"]
+        self.log.info("IO DTX committed = %s", value)
+        self.log.info("IO DTX committed min = %s", min_value)
+        self.log.info("IO DTX committed max = %s", max_value)
+        self.log.info("IO DTX committed mean = %s", mean_value)
+        self.log.info("IO DTX committed stddev = %s", stddev_value)
+
+        # DTX committed value should be ~ equal to the number of transactions
+        num_operations = block_size / transfer_size
+        self.log.info("block_size = %s", block_size)
+        self.log.info("transfer_size = %s", transfer_size)
+        self.log.info("num_operations = %s", num_operations)
+
+        return True
+
+    def test_ior_dtx_telmetry_metrics(self):
+        """JIRA ID: DAOS-7316.
+
+            Create files with transfers sizes 512 to 4M to verify the
+            DAOS engine IO DTX telemetry metrics infrastructure and
+            verify values using IOR.
+
+        :avocado: tags=all,pr,daily_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=telemetry
+        :avocado: tags=test_ior_dtx_telemetry
+
+        """
+        transfer_sizes = self.params.get("transfer_sizes", "/run/*")
+        metrics_data = {}
+        # disable verbosity
+        self.telemetry.dmg.verbose = False
+        committed_test_metrics = TelemetryUtils.ENGINE_IO_DTX_COMMITTED_METRICS
+        #committable_test_metrics = TelemetryUtils.ENGINE_IO_DTX_COMMITTABLE_METRICS
+
+        for transfer_size in transfer_sizes:
+            for operation in ["rw"]:
+                # Run ior command to populate IO dtx metrics
+                ior_results = self.ior_with_transfer_size(transfer_size, operation)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'operation' in method call
(pylint-unused-variable) Unused variable 'ior_results'